src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -475,7 +475,7 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
             MPIDI_OFI_CHOOSE_PROVIDER(prov, &prov_use, "No suitable provider provider found");
 
             /* If we picked the provider already, make sure we grab the right provider */
-            if ((NULL != provname) && (0 != strcmp(provname, prov_use->fabric_attr->prov_name))) {
+            if ((NULL != provname) && (0 != strncmp(provname, prov_use->fabric_attr->prov_name, sizeof(provname)))) {
```

Comment:
This part should be its own commit.

Reply:
The provider names are generate by the following rule "<core provider name>;<utility provider name>", i.e. "sockets;ofi_rxm", "verbs;ofi_rxm" and etc.
So, an user has to specify `MPIR_CVAR_OFI_USE_PROVIDER="<core provider>;ofi_rxm"` to ensure that correct stack of provider is selected.
In any words, if users are going to use RxM/verbs, they should specify `MPIR_CVAR_OFI_USE_PROVIDER="verbs;ofi_rxm"` (`MPIR_CVAR_OFI_USE_PROVIDER="ofi_rxm"` won't work, `MPIR_CVAR_OFI_USE_PROVIDER="verbs"` will select verbs/FI_EP_RDM provider)

I've dropped this change from the PR. And updated check for capability sets in `MPIDI_OFI_get_set_number` function -- present of "ofi_rxm" in the name of provider should be enough to chose RxM's capability set.